### PR TITLE
fix(statistics): allow all curl statistics, ordering, highlight

### PR DESCRIPTION
**Pull request by JMarkin**
Hello, If I add more statistics like 
```lua
        vim.g.rest_nvim = {
            clients = {
                curl = {
                    statistics = {
                        remote_ip = { winbar = "ip", title = "Remote IP" },
                        time_total = { winbar = "time", title = "Time" },
                        time_namelookup = { title = "Time dns" },
                        time_connect = { title = "Time connect" },
                        time_pretransfer = { title = "Time pretransfer" },
                        time_redirect = { title = "Time redirect" },
                        time_starttransfer = { title = "Time starttransfer" },
                        size_download = { title = "Download size", winbar = false },
                        speed_download = { title = "Download speed", },
                        size_upload = { title = "Upload size" },
                        speed_upload = { title = "Upload speed" },
                    },
                },
            },
        }

```
In first I got 
```
Warning  WARN [rest.nvim] Unrecognized configs found in setup: { "clients.curl.statistics.speed_download", "clients.curl.statistics.time_connect", "clients.curl.statistics.
speed_upload", "clients.curl.statistics.time_namelookup", "clients.curl.statistics.size_upload", "clients.curl.statistics.size_download.order", "clients.curl.statistics.time_total.order", "clie
nts.curl.statistics.remote_ip", "clients.curl.statistics.time_starttransfer", "clients.curl.statistics.time_redirect", "clients.curl.statistics.time_pretransfer" }
```

And after call request I got crash for winbar...
```
^I...al/share/nvim/lazy/rest.nvim/lua/rest-nvim/ui/result.lua:194: in function <...al/share/nvim/lazy/rest.nvim/lua/rest-nvim/ui/result.lua:181>
E15: Invalid expression: "v:lua.require('rest-nvim.ui.result').stat_winbar()"
2024-08-27T10:40:58 noice.nvim  ERROR vim/_editor.lua:0: nvim_exec2(): Vim(redraw):E15: Invalid expression: "v:lua.require('rest-nvim.ui.result').stat_winbar()
```

In this pr I fixed it, and add more pretty Statistics pane.

Before, default config
image

After
default config:
image

config:
```lua
        vim.g.rest_nvim = {
            clients = {
                curl = {
                    statistics = {
                        remote_ip = { winbar = "ip", title = "Remote IP", order = 1 },
                        time_total = { winbar = "time", title = "Time", order = 2 },
                        time_namelookup = { title = "Time dns", order = 3 },
                        time_connect = { title = "Time connect", order = 4 },
                        time_pretransfer = { title = "Time pretransfer", order = 5 },
                        time_redirect = { title = "Time redirect", order = 6 },
                        time_starttransfer = { title = "Time starttransfer", order = 7 },
                        size_download = { title = "Download size", winbar = false, order = 8 },
                        speed_download = { title = "Download speed", order = 9 },
                        size_upload = { title = "Upload size", order = 10 },
                        speed_upload = { title = "Upload speed", order = 11 },
                    },
                },
            },
        }

```
image

### CODE REVIEW

File: lua/rest-nvim/config/check.lua
Comment (boltlessengineer) [inline lua/rest-nvim/config/check.lua:1]: This seems too hacky. Actually I haven’t really thought about `get_unrecognized_keys` before. Now I think it is a bad idea, at least the current implementation is. It won’t work well with `table<string, T>` types.

I’ll remove it for now and reimplement later. Maybe I can use the table from `check.validate()`.

File: lua/rest-nvim/config/default.lua
Comment (boltlessengineer) [inline lua/rest-nvim/config/default.lua:1]: I’m not sure if adding a `order` property to `RestStatisticsStyle` is a good idea.
We can make `statistics` to a list if we want orders:
```lua
statistics = {
    { id = “time_total”, winbar = “take”, title = “Time taken” },
    { id = “size_download, winbar = “size”, title = “Download size” },
}
```

**Review by boltlessengineer — COMMENTED** [review]
Generally seems fair. Thank you for those fixes.

One thing I want to change is the `order` property. We can just change `statistics` table to

File: lua/rest-nvim/config/default.lua
Comment (JMarkin) [reply lua/rest-nvim/config/default.lua:1]: I fixed to skipping statistics key for format as list

**Comment by JMarkin** [comment]
@boltlessengineer what blocks this pr? Can you merge it ?

**Comment by boltlessengineer** [comment]
I’ve been busy this week. I’ll merge the PR tomorrow.

**Review by boltlessengineer — APPROVED** [review]

**Pull request merged**